src/jdk.incubator.code/share/classes/jdk/incubator/code/type/TypeVarRef.java

```diff
@@ -100,12 +102,18 @@ public JavaType erasure() {
 
     @Override
     public ExternalizedTypeElement externalize() {
-        return new ExternalizedTypeElement(String.format("#%s::%s", owner, name),
-                List.of(bound.externalize()));
+        ExternalizedTypeElement eOwner = switch (owner) {
+            // @@@ Should be able to use single case matching TypeElement
+            case JavaRef ref -> ref.externalize();
+            case ClassType classType -> classType.externalize();
+        };
+        return new ExternalizedTypeElement("#" + name,
+                List.of(eOwner, bound.externalize()));
     }
 
     @Override
     public String toString() {
+        // @@@ required to pass TestJavaType.java
         return name;
```

Comment:
What should the `toString()` method in JavaType do? Should they return a human readable representation that will then be embedded in the code model (e.g. when calling `toText`) ? Or should we also have `toText` on all `TypeElement`s ?

Reply:
Hmm... in this case `toString()` seems appropriate to return a human readable representation (that `OpWriter` writes and `OpParser` parses, but we don't have to specify the grammar), given that i would not expect these strings to be very large, unlike the text of code models.

(The comment indicates some testing modifications would be required for type variable types.)
The default implementation of `TypeElement.toString` could be `return externalize().toString()`.

Reply:
The issue is that there is a clear tension between things being parseable and things being readable. If you want things to be parseable, you need to include all the information that is in the ref/type. For instance, a type-variable should have the owner printed out -- as well as the bound. But for a type-variable, a human readable representation would just expect to see the tvar name (and maybe the owner, at least in some ambiguous cases). But such a representation would be lossy, and not parseable.